aggregator/priceNotifier.go

```diff
@@ -61,6 +62,12 @@ func NewPriceNotifier(args ArgsPriceNotifier) (*priceNotifier, error) {
 		return nil, err
 	}
 
+	for _, argsPair := range args.Pairs {
+		denominationFactorAsFloat64 := math.Pow(10, float64(argsPair.Decimals))
+		argsPair.DenominationFactor = uint64(denominationFactorAsFloat64)
```

Comment:
I do not like the pattern of changing received arguments.
We should either pass the Decimals further on to the sub components or somehow cache the 2 values DenominationFactor and TrimPrecision in another internal structure

Reply:
refactored